### PR TITLE
Getting started with Gradient descent algorithm in python

### LINKED ISSUE #4138: [Machine learning] Getting started with Gradient Descent algorithm
#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximize the chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

#### Proposed title of the article
Getting started with Gradient Descent algorithm

Use keyword research to improve your article's SEO.

#### Proposed article introduction
Gradient descent is an optimization algorithm that is used when training a machine learning model. It's based on a convex function and tweaks its parameters iteratively to minimize a given function to its local minimum. This algorithm is widely used in machine learning and deep learning to optimize the parameters W( weights) and b(bias term) for the entire training set.

In this article, we shall learn how the gradient descent algorithm is developed and how it works. Then, using linear regression weights, we shall carry out a python implementation of gradient descent.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
1. Learner will clearly understand how gradient descent work.
2. Later, the learner will be able to develop a gradient descent algorithm independently.
3. Learners will learn how to implement the developed algorithm in no. 2 above; hence they will be able to implement any future algorithm in python.

#### Article quality
This article covers not only the intuition part but also the implementation of our optimization algorithm.

#### References
N/A

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by jacksonjate1**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4138

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @jacksonjate1 

Kindly add an author page for your article. You must have missed this.

**Review by mercymeave — APPROVED** [review]
Thank you @jacksonjate1 for submitting such a great article!

Let us wait for a final reviewer to finalize this PR.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@mercymeave 

Please go through my edits as there were various missing things while I did my Final Review. 

There were entire sentences missing! 🤔 🔍  

cc @jacksonjate1  Please do the same - an article of this caliber should have been closed asap - to allow you the time to make the proper edits. 
 
There were definitions missing, incomplete sentences, and misspellings.

If you TWO could please go through my edits to ensure I did not deleted anything that was needed - or changed the meaning of any content.
